libpromises/attributes.c

```diff
@@ -1261,10 +1265,17 @@ NewPackages GetNewPackageConstraints(const EvalContext *ctx, const Promise *pp)
 
         VarRef *ref = VarRefParseFromScope("package_module_knowledge.platform_default", NULL);
```

Comment:
Should this be looking at `package_module` in `body common control`  (which the MPF defaults to the value of `package_module_knowledge.platform_default`? The control body attributes are exposed as variables ...

`default:control_common.package_module`

Reply:
Also, I wonder .. parse from scope ... If the package promise were in a non-default namespace, seems like this might be checking the value of `MyNonDefaultNamespace:package_module_knowledge.platform_default`

Reply:
Right, this wasn't a part of this change but rather a previous change that introduced the use of `package_module_knowledge.platform_default` but I agree it should be changed to something as close to the core as possible. After I fix that I will ask our bot to generate packages for testing.